docs/gcloud-configuration.md

```diff
@@ -0,0 +1,38 @@
+# Gcloud Configuration
+
+You can configure Google Cloud Monitoring Uptime Checks as a Ingress Monitor by using below configuration:
+
+| Key          | Description                                                                         |
+| -------------| ----------------------------------------------------------------------------------- |
+| name         | Name of the provider (e.g. gcloud)                                                  |
+| apiKey       | JSON Service Account Key                                                            |
+| gcloudConfig | `gcloudConfig` is the configuration specific to gcloud Instance as mentioned below: |
+
+## gcloud Configuration:
+
+| Key       | Description                                                                                                    |
+| --------- | ----------------------- |
+| projectId | Google Cloud Project ID |
+
+**Example Configuration:**
+
+```yaml
+providers:
+  - name: gcloud
+    apiKey: |
+      {
+        "type": "service_account",
+        "project_id": "...",
+        "private_key_id": "...",
+        "private_key": "-----BEGIN PRIVATE KEY-----\n...\n-----END PRIVATE KEY-----\n",
+        "client_email": "...iam.gserviceaccount.com",
+        "client_id": "...",
+        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+        "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/...iam.gserviceaccount.com"
+      }
```

Comment:
Not too familiar with gcloud. Do you think we need the whole block for authentication?

Reply:
Gcloud Provides Keys in JSON format. This is the default way to authenticate a service account.

https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating_service_account_keys